api/v1alpha1/backend.go

```diff
@@ -2,147 +2,149 @@ package v1alpha1
 
 import (
 	"fmt"
+	"reflect"
+
 	"github.com/vshn/k8up/cfg"
 
 	corev1 "k8s.io/api/core/v1"
 )
 
-type Backend struct {
-	// RepoPasswordSecretRef references a secret key to look up the restic repository password
-	RepoPasswordSecretRef *corev1.SecretKeySelector `json:"repoPasswordSecretRef,omitempty"`
-	Local                 *LocalSpec                `json:"local,omitempty"`
-	S3                    *S3Spec                   `json:"s3,omitempty"`
-	GCS                   *GCSSpec                  `json:"gcs,omitempty"`
-	Azure                 *AzureSpec                `json:"azure,omitempty"`
-	Swift                 *SwiftSpec                `json:"swift,omitempty"`
-	B2                    *B2Spec                   `json:"b2,omitempty"`
-	Rest                  *RestServerSpec           `json:"rest,omitempty"`
-}
-
-// GetCredentialEnv will return a map containing the correct
-func (b *Backend) GetCredentialEnv() map[string]*corev1.EnvVarSource {
-	vars := make(map[string]*corev1.EnvVarSource)
-
-	if b.RepoPasswordSecretRef != nil {
-		vars[cfg.ResticPasswordEnvName] = &corev1.EnvVarSource{
-			SecretKeyRef: b.RepoPasswordSecretRef,
-		}
-	}
-
-	if b.Azure != nil {
-		return b.Azure.EnvVars(vars)
-	}
-
-	if b.B2 != nil {
-		return b.B2.EnvVars(vars)
+type (
+	// Backend allows configuring several backend implementations.
+	// It is expected that users only configure one storage type.
```

Comment:
We should probably document that expectation somewhere, if that hasn't happened already.

Reply:
Nope, in fact there's barely any backends documented.